python/lsst/validate/drp/validate.py

```diff
@@ -0,0 +1,368 @@
+#!/usr/bin/env python
+
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+from __future__ import print_function, absolute_import
+
+import numpy as np
+import yaml
+
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+from lsst.afw.table import SourceCatalog, SchemaMapper, Field
+from lsst.afw.table import MultiMatch, SimpleRecord, GroupView
+import lsst.daf.persistence as dafPersist
+import lsst.pipe.base as pipeBase
+
+from .calcSrd import calcAM1, calcAM2
+from .check import checkAstrometry, checkPhotometry, positionRms
+from .print import printPA1, printPA2, printAM1, printAM2
+from .plot import plotAstrometry, plotPhotometry, plotPA1, plotAM1, plotAM2
+from .util import getCcdKeyName, repoNameToPrefix
+
+
+def loadDataIdsAndParameters(configFile):
+    """Load data IDs, magnitude range, and expected metrics from a yaml file.
+
+    Inputs
+    ------
+    configFile : str
+        YAML file that stores visit, filter, ccd,
+        good_mag_limit, medianAstromscatterRef, medianPhotoscatterRef, matchRef
+
+    Returns
+    -------
+    dict, float, float, float
+        dataIds, good_mag_limit, medianRef, matchRef
+    """
+    stream = open(configFile, mode='r')
+    data = yaml.load(stream)
+
+    ccdKeyName = getCcdKeyName(data)
+    visitDataIds = constructDataIds(data['visits'], data['filter'],
+                                    data[ccdKeyName], ccdKeyName)
+
+    return (visitDataIds,
+            data['good_mag_limit'],
+            data['medianAstromscatterRef'],
+            data['medianPhotoscatterRef'],
+            data['matchRef'],
+           )
+
+
+def constructDataIds(visits, filter, ccds, ccdKeyName=None):
+    """Construct a comprehensive set of Butler Data IDs for a repository.
```

Comment:
Improved docstring.

If some CCDs don't exist that will be a failure of some step of the processing.

DM-5120 will parse repos and extract the list of useful dataIds therein.

Thanks for the re-ordering of the arguments.  Definitely agreed.